python/tvm/auto_scheduler/task_scheduler.py

```diff
@@ -0,0 +1,452 @@
+    strategy: Optional[str]
+        The scheduling strategy.
+        "round-robin": Tune tasks in round robin order.
+        "gradient" : Tune tasks with gradient descent.
+    load_model_file: Optional[str]
+        Load pre-trained model from this file
+    load_log_file: Optional[str]
+        Load measurement records from this file
+    eps_random: float = 0.05
+        Always allocate this percent of n_trials to select tasks randomly.
+        This is for encouraging exploration.
+    verbose: int = 1
+        The level of verbosity. 0 means silent.
+    alpha: float = 0.2
+        The parameter used for 'gradient' strategy
+    beta: float = 2
+        The parameter used for 'gradient' strategy
+    backward_window_size: int = 3
+        The parameter used for 'gradient' strategy
+    """
+
+    def __init__(
+        self,
+        tasks,
+        objective_func,
+        strategy="gradient",
+        load_model_file: str = None,
+        load_log_file: str = None,
+        eps_random: float = 0.05,
+        verbose: int = 1,
+        alpha: float = 0.2,
+        beta: float = 2,
+        gamma: float = 0.5,
+        backward_window_size: int = 3,
+    ):
+        super().__init__(tasks, objective_func)
+        self.strategy = strategy
+        self.eps_random = eps_random
+        self.verbose = verbose
+        self.load_log_file = load_log_file
+        self.load_model_file = load_model_file
+        self.alpha = alpha
+        self.beta = beta
+        self.gamma = gamma
+        self.backward_window_size = backward_window_size
+
+        assert self.strategy in ["round-robin", "gradient"]
+
+        # task_cts[i] saves how many times task i is tuned
+        self.task_cts = [0 for _ in range(len(self.tasks))]
+
+        # task_costs_history[i] saves the latency history of task i
+        self.task_costs_history = [[] for _ in range(len(self.tasks))]
+
+        # best_costs[i] saves the best latency of task i
+        self.best_costs = 1e10 * np.ones(len(self.tasks))
+
+        self.tune_option = self.measurer = self.search_policies = self.ct = self.tic = None
+        self.num_measures_per_round = None
+        self.dead_tasks = set()
+        self.sequential_now_task_idx = 0
+        self.sequential_now_task_begin_ct = 0
+
+        assert len(tasks) != 0, "No tasks"
+
+        # Build similarity group
+        self.task_tags = []
+        self.tag_to_group_id = {}
+        self.group_task_ids = []
+        self.flop_cts = []
+        for i, task in enumerate(self.tasks):
+            tag = derive_similarity_tag(task.compute_dag)
+            self.task_tags.append(tag)
+            self.flop_cts.append(task.compute_dag.flop_ct)
+            if tag == "":
+                continue
+
+            if tag not in self.tag_to_group_id:
+                self.tag_to_group_id[tag] = len(self.tag_to_group_id)
+                self.group_task_ids.append([])
+            self.group_task_ids[self.tag_to_group_id[tag]].append(i)
```

Comment:
It seems they are equivalent. I added some notes to make the code easier to understand.
However, I want to avoid introducing bugs accidentally so I don't change the code.